doc/api/modules.md

```diff
@@ -63,6 +63,17 @@ module.exports = class Square {
 
 The module system is implemented in the `require('module')` module.
 
+## Enabling
+
+<!-- type=misc -->
+
+Node.js treats JavaScript code as CommonJS modules by default.
+Authors can tell Node.js to treat JavaScript code as CommonJS modules
+via the `.cjs` file extension, the `package.json` [`"type"`][] field, or the
+`--input-type` flag. See
+[Determining module system](packages.md#determining-module-system) for more
+details.
```

Comment:
```suggestion
Node.js has two module systems: CommonJS and ES modules.
Node.js treats JavaScript code as CommonJS modules by default.
Within an ES module context, authors can tell Node.js to treat
JavaScript code as CommonJS modules via the `.cjs` file extension,
the `package.json` [`"type"`][] field, or the `--input-type` flag. See
[Determining module system](packages.md#determining-module-system) for more
details.
```

Reply:
Everything is an es module context. Type module doesn’t create an “es module context”, it just changes the default parse goal for .js files.

Reply:
I guess, put another way - what exactly is an “ES module context”? Within a filesystem, you can use extensions with or without the type module flag; in a command line, you can use the input-type flag.

Reply:
I'm not sure I understand what would an ES module context mean either. Also, FWIW I've copied this section from `esm.md`, so if we want to change its wording maybe it's best to make a separate PR for that? I'm tempted to remove this change from this PR so it can be discussed separately. wdyt?https://github.com/nodejs/node/blob/2bc1d9245b987aa747173f5aea4883df18314f46/doc/api/esm.md#L97-L102